File: app/services/fleetio/vehicle.rb
module Fleetio
  class Vehicle < Base
    attr_accessor :color,
                  :make,
                  :model,
                  :year,
                  :name,
                  :license_plate,
                  :current_meter_value,
                  :vin

    def initialize(args = {})
      super(args)
    end

    def self.find(vin)
      response = Request.get("vehicles?q[vin_eq]=#{vin}")
      response_hash = response[0].to_h
      scoped_response = response_hash.slice("color", "make", "model", "year", "name",
                                            "license_plate", "current_meter_value", "vin",
                                            "group_name", "vehicle_status_color",
                                            "registration_state", "vehicle_status_name",
                                            "default_image_url_medium",
                                            "default_image_url_large")
    end   
  end
end

File: client/src/components/Vehicle.js
import React from 'react';
import { Icon, Table, Popup, Grid, Image } from 'semantic-ui-react';

const Vehicle = ({vehicle}) =>
  <Popup trigger={	
  	<Table.Row>
		<Table.Cell className="single-vehicle" key={vehicle.id}>
			<h4><Icon color={vehicle.vehicle_status_color} name='circle' />&nbsp;&nbsp;{vehicle.vehicle_status_name}</h4>
		</Table.Cell>
		<Table.Cell>
			<h4>{vehicle.name}</h4>
		</Table.Cell>		
		<Table.Cell>
			<h4>{vehicle.make}</h4>
		</Table.Cell>
		<Table.Cell>
			<h4>{vehicle.model}</h4>
		</Table.Cell>		
		<Table.Cell>
			<h4>{vehicle.color}</h4>
		</Table.Cell>
		<Table.Cell>
			<h4>{vehicle.year}</h4>
		</Table.Cell>
	</Table.Row> } flowing hoverable>
	  <Grid columns='three' divided>
	    <Grid.Row>
	      <Grid.Column>
	      	<h4>{vehicle.name}</h4>
	      	<Image src={vehicle.default_image_url_medium} circular/>
	      </Grid.Column>
	      <Grid.Column>
	      	<h4>Odometer:</h4>
	      	{vehicle.current_meter_value}
	      </Grid.Column>	      
	      <Grid.Column>
	      	<h4>Group Name:</h4>
	      	{vehicle.group_name}
	      </Grid.Column>	      
	    </Grid.Row>
	    <Grid.Row>
	      <Grid.Column>
	      	<h4>VIN:</h4>
	      	{vehicle.vin}
	      </Grid.Column>	    
	      <Grid.Column>
	      	<h4>License Plate:</h4>
	      	{vehicle.license_plate}
	      </Grid.Column>
	      <Grid.Column>
	      	<h4>Registered State:</h4>
	      	{vehicle.registration_state}
	      </Grid.Column>	      
	    </Grid.Row>
	  </Grid>
  </Popup>
export default Vehicle;